### PR TITLE
Fix Broken Stream Close in writeRawValue

### PR DESCRIPTION
**Pull request by original-brownbear**
Small oversight in #56078 that only showed up during backporting where a stream copy was turned from a non-closing to a closing one. Enhanced part of a test in this PR to make it show up in `master` also even though we practically never use this method with stream targets that actually `close`.

### CODE REVIEW

**Review by rjernst — APPROVED** [review]
LGTM

File: server/src/test/java/org/elasticsearch/common/xcontent/BaseXContentTestCase.java [hunk 1]
```diff
@@ -947,11 +948,18 @@ void doTestRawValue(XContent source) throws Exception {
...
+            @Override
+            public void close() {
+                closed.set(true);
+            }
+        };
         try (XContentGenerator generator = xcontentType().xContent().createGenerator(os)) {
             generator.writeStartObject();
             generator.writeFieldName("test");
             generator.writeRawValue(new BytesArray(rawData).streamInput(), source.type());
+            assertFalse(closed.get());
```
Comment (pugnascotia) [inline server/src/test/java/org/elasticsearch/common/xcontent/BaseXContentTestCase.java:962]: What do you think about adding a message here? e.g.

```suggestion
            assertFalse("Generator should not have close the output stream", closed.get());
```

Comment (original-brownbear) [reply server/src/test/java/org/elasticsearch/common/xcontent/BaseXContentTestCase.java:962]: ++

**Review by pugnascotia — APPROVED** [review]

**Comment by original-brownbear** [comment]
BwC tests are failing because of https://github.com/elastic/elasticsearch/pull/60641

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/bwc
Jenkins run elasticsearch-ci/default-distro

**Comment by original-brownbear** [comment]
Thanks Ryan + Rory!

**Pull request merged**